nodeJS/express/introduction_to_express.md

```diff
@@ -0,0 +1,98 @@
+
+In this course, we are going to be using a backend framework called [Express](https://expressjs.com/). It will handle some of the more cumbersome things for us, and make the things we did previously far simpler. Express itself is an intentionally barebones and unopinionated framework; it allows us to do many things how we want, and to extend it with only the features we need. While this gives us great flexibility in how we do things, it can be a little tricky deciding between multiple viable options at times.
+
+Going forward, we will be diving into how we can use Express in various ways, such as to create a full stack application utilising the `Model View Controller` (MVC) pattern, as well as to create a REST API just like ones you will have used before for things such as the Weather App project or the React Shopping Cart. Along the way, we will be learning about things like databases, authentication, and much more. There is a lot to take in, so take it steady and do not be afraid to ask our community for help in [our Discord server](https://discord.gg/theodinproject)!
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Describe what Express is.
+- Describe the relationship between Express and Node.
+- Set up a basic Express server.
+- Describe how an incoming request is processed and ultimately responded to by the server.
+- Describe what a middleware function is.
+- Describe how to use `nodemon` to automatically restart your application when a change is made.
+
+### Setting up Express
+
+Let's make a basic Express app. Inside a new directory, start by running `npm init` and filling in each option to create a `package.json`. Once that's created, we can install the Express dependency.
+
+```bash
+npm install express
+```
+
+We can now create an `app.js` file that will serve as the starting point for our Express server. We could name this file anything we want, such as `index.js` or `main.js`, but we will use the same name used in the Express documentation. Inside `app.js`, we will have the following:
+
+```javascript
+const express = require("express");
+const app = express();
+
+app.get("/", (req, res) => res.send("Hello, world!"));
+
+const PORT = 3000;
+app.listen(PORT, () => console.log(`My first Express app - listening on port ${PORT}!`));
+```
+
+Let's break this down. Firstly, we import `express` using `require()`, then we call it to create the actual server that we store in the `app` variable. While you can technically do this here in one line via `const app = require("express")();`, very often you will need features that come from properties on `express`. Many of these features will be covered in later lessons.
```

Comment:
Feels to verbose, and not sure if we need to add the `require('express')();` thing. Wording suggestion:
```suggestion
Let's break this down. We import and call `express` to initialize the `app` variable.  This is our server.
```

Reply:
In fairness, not the greatest show but I actually completely forgot I wrote this extra `require()()` stuff in. The initial idea was to answer "bu why do we need an intermediary variable?" but in hindsight, don't think that'd be a common question and anyone who does ask that in the community can just be asnwered that/told to keep going.